mathics/core/pattern.py

```diff
@@ -334,7 +334,6 @@ def yield_head(head_vars, _):
 
     def get_pre_choices(self, yield_func, expression, attributes, vars):
         if "System`Orderless" in attributes:
-            self.sort()
```

Comment:
Why do you think this can be removed?

Reply:
If that is orderless, why does the order matter? And the tests pass.

Reply:
So maybe that should be if not "System`Orderless". 

A couple of important things. 

1. Don't assume that if test passing means that code is correct
2. If you see code that looks funny and more complicated make an effort to find out _why_ that code was added in the first place. Git blame might help here.

Reply:
Yes, canonicalizing before pattern matching was the idea here AFAIR.